src/main/java/io/github/nucleuspowered/nucleus/modules/environment/commands/WeatherCommand.java

```diff
@@ -88,6 +101,11 @@ public CommandResult executeCommand(CommandSource src, CommandContext args) thro
             oi =  i.isPresent() ? Optional.of((long)i.get()) : Optional.empty();
         }
 
+        if (oi.orElse(Long.MAX_VALUE) > eca.getNodeOrDefault().getMaximumWeatherTimespan() && !permissions.testSuffix(src, "exempt.length") && eca.getNodeOrDefault().getMaximumWeatherTimespan() != -1) {
```

Comment:
We probably want to catch people using 0 or any negative number, so check for > 0 rather than not -1.